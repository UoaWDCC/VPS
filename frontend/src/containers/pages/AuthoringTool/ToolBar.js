import React, { useState } from "react";
import {
  Box,
  Tooltip,
  Button,
  MenuList,
  Menu,
  MenuItem,
} from "@material-ui/core";
import ImageIcon from "@material-ui/icons/Image";
import TextFieldsIcon from "@material-ui/icons/TextFields";
import VolumeUpIcon from "@material-ui/icons/VolumeUp";
import styles from "../../../styling/ToolBar.module.scss";

const SidebarData = [
  {
    title: "Background Image",
    icon: <ImageIcon fontSize="medium" />,
    dropdown: {},
    onClick: null,
  },
  {
    title: "Text",
    icon: <TextFieldsIcon fontSize="medium" />,
    dropdown: {},
    onClick: null,
  },
  {
    title: "Audio",
    icon: <VolumeUpIcon fontSize="medium" />,
    dropdown: null,
    onClick: null,
  },
];

export default function ToolBar() {
  const [anchorEl, setAnchorEl] = useState(null);
  const handleDropdownClick = (e) => {
    setAnchorEl(e.currentTarget);
  };

  const handleDropdownClose = () => {
    setAnchorEl(null);
  };
  return (
    <>
      <MenuList className={styles.toolBar}>
        <Box display="flex" flexDirection="row">
          {SidebarData.map((val) => {
            const menuOnClick = val.dropdown
              ? handleDropdownClick
              : val.onClick;
            const open = val.title === anchorEl?.getAttribute("title");
            return (
              <>
                <Tooltip title={val.title} enterDelay={200} key={val.title}>
                  <Button
                    className={styles.menuItem}
                    key={val.title}
                    onClick={menuOnClick}
                  >
                    {val.icon}
                  </Button>
                </Tooltip>
                <Menu
                  id={val.title}
                  open={open}
                  anchorEl={anchorEl}
                  getContentAnchorEl={null}
                  onClose={handleDropdownClose}
                  anchorOrigin={{
                    vertical: "bottom",
                    horizontal: "left",
                  }}
                  transformOrigin={{
                    vertical: "top",
                    horizontal: "left",
                  }}
                >
                  <MenuItem onClick={() => {}}>Profile</MenuItem>
                </Menu>
              </>
            );
          })}
        </Box>
      </MenuList>
    </>
  );
}
