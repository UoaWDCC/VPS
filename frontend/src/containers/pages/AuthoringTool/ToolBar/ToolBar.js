import React, { useState, useContext } from "react";
import { Box, Tooltip, Button, MenuList, Menu } from "@material-ui/core";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import toolBarData from "./ToolBarData";
import SceneContext from "../../../../context/SceneContext";
import styles from "../../../../styling/ToolBar.module.scss";

export default function ToolBar() {
  const { currentScene, setCurrentScene } = useContext(SceneContext);
  const [anchorEl, setAnchorEl] = useState(null);
  const [triggerElTitle, setTriggerElTitle] = useState(null);
  const handleDropdownClick = (e, title) => {
    setTriggerElTitle(title);
    setAnchorEl(e.currentTarget);
  };

  const handleDropdownClose = () => {
    setTriggerElTitle(null);
    setAnchorEl(null);
  };
  return (
    <>
      <MenuList className={styles.toolBar}>
        <Box display="flex" flexDirection="row">
          {toolBarData.map((tool) => {
            const menuOnClick = tool.dropdown
              ? (event) => {
                  handleDropdownClick(event, tool.title);
                }
              : () => tool.onClick(currentScene, setCurrentScene);
            const open = tool.title === triggerElTitle;
            return (
              <div key={tool.title}>
                <Tooltip title={tool.title} enterDelay={200} key={tool.title}>
                  <Button
                    className={styles.menuItem}
                    key={tool.title}
                    onClick={menuOnClick}
                  >
                    {tool.icon}
                    {tool.dropdown && <ArrowDropDownIcon fontSize="small" />}
                  </Button>
                </Tooltip>
                {tool.dropdown && (
                  <SubMenu
                    tool={tool}
                    open={open}
                    anchorEl={anchorEl}
                    handleDropdownClose={handleDropdownClose}
                  />
                )}
              </div>
            );
          })}
        </Box>
      </MenuList>
    </>
  );
}

// This is the tool dropdown
// It is only rendered when the tool component has a dropdown
const SubMenu = ({ tool, open, anchorEl, handleDropdownClose }) => {
  return (
    <Menu
      id={tool.title}
      open={open}
      anchorEl={anchorEl}
      getContentAnchorEl={null}
      onClose={handleDropdownClose}
      anchorOrigin={{
        vertical: "bottom",
        horizontal: "left",
      }}
      transformOrigin={{
        vertical: "top",
        horizontal: "left",
      }}
    >
      {tool.dropdown.map((dropdown, index) => {
        // eslint-disable-next-line react/no-array-index-key
        return <div key={index}>{dropdown.component}</div>;
      })}
    </Menu>
  );
};
