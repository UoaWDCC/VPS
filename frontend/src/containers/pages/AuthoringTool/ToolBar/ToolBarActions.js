import { v4 } from "uuid";

function addButton(currentScene, setCurrentScene) {
  const newButton = {
    type: "BUTTON",
    text: "Button",
    variant: "contained",
    colour: "white",
    nextScene: "",
    left: 0, // as percentage
    top: 0, // as percentage
    height: 6, // as percentage
    width: 20, // as percentage
    id: v4(),
  };

  const updatedComponents = currentScene.components;

  updatedComponents.push(newButton);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

function addText(currentScene, setCurrentScene) {
  const newText = {
    type: "TEXT",
    text: "default text",
    border: true,
    fontSize: 16, // pt
    color: "black",
    textAlign: "left",
    left: 0, // as percentage
    top: 0, // as percentage
    height: 10, // as percentage
    width: 20, // as percentage
    id: v4(),
  };

  const updatedComponents = currentScene.components;

  updatedComponents.push(newText);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

function addImage(currentScene, setCurrentScene, image) {
  const newImage = {
    type: "IMAGE",
    imageId: image._id,
    left: 0, // as percentage
    top: 0, // as percentage
    height: "auto", // as percentage
    width: "auto", // as percentage
    id: v4(),
  };

  const updatedComponents = currentScene.components;

  updatedComponents.push(newImage);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

function addFirebaseImage(currentScene, setCurrentScene, fileObject, url) {
  const newImage = {
    type: "FIREBASEIMAGE",
    fileObject,
    url,
    left: 0, // as percentage
    top: 0, // as percentage
    height: "auto", // as percentage
    width: "auto", // as percentage
    id: v4(),
  };

  const updatedComponents = currentScene.components;

  updatedComponents.push(newImage);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

function addFirebaseAudio(currentScene, setCurrentScene, fileObject, url) {
  const newAudio = {
    type: "FIREBASEAUDIO",
    name: fileObject.name,
    fileObject,
    url,
    loop: false,
    left: 0, // as percentage
    top: 0, // as percentage
    height: 10, // as percentage
    width: "auto", // icon ratio is fixed
    id: v4(),
  };

  const updatedComponents = currentScene.components;

  updatedComponents.push(newAudio);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

export { addButton, addText, addImage, addFirebaseImage, addFirebaseAudio };
