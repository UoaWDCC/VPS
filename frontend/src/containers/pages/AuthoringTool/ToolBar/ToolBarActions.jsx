import { v4 } from "uuid";

/**
 * @param {object} currentScene
 * @param {function} setCurrentScene
 */
function addComponent(component, currentScene, setCurrentScene) {
  const updatedComponents = currentScene.components;

  updatedComponents.push(component);

  setCurrentScene({
    ...currentScene,
    components: updatedComponents,
  });
}

/**
 * function to be put into ToolBarData when button is added
 */
function addButton(currentScene, setCurrentScene) {
  const newButton = {
    type: "BUTTON",
    text: "Button",
    variant: "contained",
    colour: "white",
    nextScene: "",
    left: 0, // as percentage
    top: 0, // as percentage
    height: 6, // as percentage
    width: 20, // as percentage
    id: v4(),
  };

  addComponent(newButton, currentScene, setCurrentScene);
}

/**
 * function to be put into ToolBarData when text is added
 */
function addText(currentScene, setCurrentScene) {
  const newText = {
    type: "TEXT",
    text: "default text",
    border: true,
    fontSize: 16, // pt
    color: "black",
    textAlign: "left",
    left: 0, // as percentage
    top: 0, // as percentage
    height: 10, // as percentage
    width: 20, // as percentage
    id: v4(),
  };

  addComponent(newText, currentScene, setCurrentScene);
}

function addSpeechText(currentScene, setCurrentScene) {
  const newSpeechText = {
    type: "SPEECHTEXT",
    text: "default speech text",
    border: true,
    fontSize: 16, // pt
    color: "black",
    textAlign: "left",
    left: 0, // as percentage
    top: 0, // as percentage
    height: 10, // as percentage
    width: 20, // as percentage
    arrowLocation: "top",
    id: v4(),
  };

  addComponent(newSpeechText, currentScene, setCurrentScene);
}

/**
 * function to be put into ToolBarData when image is added
 * @param {object} image
 */
function addImage(currentScene, setCurrentScene, image) {
  const newImage = {
    type: "IMAGE",
    imageId: image._id,
    left: 0, // as percentage
    top: 0, // as percentage
    height: "auto", // as percentage
    width: "auto", // as percentage
    id: v4(),
  };

  addComponent(newImage, currentScene, setCurrentScene);
}

/**
 * function to be put into ToolBarData when firebase image is added
 * @param {object} fileObject
 * @param {string} url
 */
function addFirebaseImage(currentScene, setCurrentScene, fileObject, url) {
  const newImage = {
    type: "FIREBASEIMAGE",
    fileObject,
    url,
    left: 0, // as percentage
    top: 0, // as percentage
    height: "auto", // as percentage
    width: "auto", // as percentage
    id: v4(),
  };

  addComponent(newImage, currentScene, setCurrentScene);
}

/**
 * call this when user adds firebase audio
 *
 * @param {object} fileObject
 * @param {string} url
 */
function addFirebaseAudio(currentScene, setCurrentScene, fileObject, url) {
  const newAudio = {
    type: "FIREBASEAUDIO",
    name: fileObject.name,
    fileObject,
    url,
    loop: false,
    left: 0, // as percentage
    top: 0, // as percentage
    height: 10, // as percentage
    width: 5, // as percentage
    id: v4(),
  };

  addComponent(newAudio, currentScene, setCurrentScene);
}

export {
  addButton,
  addFirebaseAudio,
  addFirebaseImage,
  addImage,
  addSpeechText,
  addText,
};
