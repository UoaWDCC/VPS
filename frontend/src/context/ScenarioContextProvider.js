import React, { useState } from "react";
import { useGet } from "../hooks/crudHooks";
import useLocalStorage from "../hooks/useLocalStorage";
import ScenarioContext from "./ScenarioContext";

export default function ScenarioContextProvider({ children }) {
  const [currentScenario, setCurrentScenario] = useLocalStorage(
    "currentScenario",
    null
  );
  const [scenarios, setScenarios] = useState();
  const { reFetch } = useGet(`api/scenario`, setScenarios);

  return (
    <ScenarioContext.Provider
      value={{
        scenarios,
        setScenarios,
        reFetch,
        currentScenario,
        setCurrentScenario,
      }}
    >
      {children}
    </ScenarioContext.Provider>
  );
}
