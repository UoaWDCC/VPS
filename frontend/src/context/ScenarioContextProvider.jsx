import { useState } from "react";
import { useGet } from "../hooks/crudHooks";
import useLocalStorage from "../hooks/useLocalStorage";
import ScenarioContext from "./ScenarioContext";

/**
 * This is a Context Provider made with the React Context API
 * ScenarioContextProvider allows access to scenario info and the refetch function
 */
export default function ScenarioContextProvider({ children }) {
  const [currentScenario, setCurrentScenario] = useLocalStorage(
    "currentScenario",
    null
  );
  const [scenarios, setScenarios] = useState();
  const [assignedScenarios, setAssignedScenarios] = useState();
  const [roleList, setRoleList] = useState();

  const { reFetch } = useGet(`api/scenario`, setScenarios);
  const { reFetch: reFetch2 } = useGet(
    `api/scenario/assigned`,
    setAssignedScenarios
  );

  const { reFetch: reFetch3 } = useGet(
    currentScenario ? `api/group/${currentScenario._id}/roleList` : null,
    currentScenario ? setRoleList : () => {}
  );

  return (
    <ScenarioContext.Provider
      value={{
        scenarios,
        setScenarios,
        reFetch,
        assignedScenarios,
        setAssignedScenarios,
        reFetch2,
        reFetch3,
        currentScenario,
        setCurrentScenario,
        roleList,
      }}
    >
      {children}
    </ScenarioContext.Provider>
  );
}
