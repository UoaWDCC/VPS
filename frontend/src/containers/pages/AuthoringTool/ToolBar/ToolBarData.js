import React from "react";
import ImageIcon from "@material-ui/icons/Image";
import TextFieldsIcon from "@material-ui/icons/TextFields";
import ButtonIcon from "@material-ui/icons/AddCircle";
import VolumeUpIcon from "@material-ui/icons/VolumeUp";
import ChatBubbleIcon from "@material-ui/icons/ChatBubble";
import ChooseBackgroundSubMenu from "./Background/ChooseBackgroundSubMenu";
import { addButton, addText, addSpeechText } from "./ToolBarActions";
import UploadImage from "./Background/UploadImage";
import UploadAudio from "./Audio/UploadAudio";
import PlayersListSubMenu from "./Background/PlayersListSubMenu";

/**
 * This file contains the data for the add component icons to be added into the ToolBar
 */
const toolBarData = [
  {
    title: "Image",
    icon: <ImageIcon fontSize="medium" />,
    dropdown: [
      {
        component: <ChooseBackgroundSubMenu />,
      },
      {
        component: <UploadImage />,
      },
    ],
  },
  {
    title: "Text",
    icon: <TextFieldsIcon fontSize="medium" />,
    onClick: addText,
  },
  {
    title: "Button",
    icon: <ButtonIcon fontSize="medium" />,
    onClick: addButton,
  },
  {
    title: "Audio",
    icon: <VolumeUpIcon fontSize="medium" />,
    dropdown: [
      {
        component: <UploadAudio />,
      },
    ],
  },
  {
    title: "Speech text",
    icon: <ChatBubbleIcon fontSize="medium" />,
    onClick: addSpeechText,
  },
  {
    title: "Apply Tags",
    icon: <ImageIcon fontSize="medium" />,
    dropdown: [
      {
        component: <PlayersListSubMenu />,
      },
    ],
  },
];

export default toolBarData;
