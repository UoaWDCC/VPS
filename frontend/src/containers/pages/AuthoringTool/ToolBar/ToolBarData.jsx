import ButtonIcon from "@material-ui/icons/AddCircle";
import ChatBubbleIcon from "@material-ui/icons/ChatBubble";
import ImageIcon from "@material-ui/icons/Image";
import TextFieldsIcon from "@material-ui/icons/TextFields";
import VolumeUpIcon from "@material-ui/icons/VolumeUp";
import UploadAudio from "./Audio/UploadAudio";
import ChooseBackgroundSubMenu from "./Background/ChooseBackgroundSubMenu";
import UploadImage from "./Background/UploadImage";
import { addButton, addSpeechText, addText } from "./ToolBarActions";

/**
 * This file contains the data for the add component icons to be added into the ToolBar
 */
const toolBarData = [
  {
    title: "Image",
    icon: <ImageIcon fontSize="medium" />,
    dropdown: [
      {
        component: <ChooseBackgroundSubMenu />,
      },
      {
        component: <UploadImage />,
      },
    ],
  },
  {
    title: "Text",
    icon: <TextFieldsIcon fontSize="medium" />,
    onClick: addText,
  },
  {
    title: "Button",
    icon: <ButtonIcon fontSize="medium" />,
    onClick: addButton,
  },
  {
    title: "Audio",
    icon: <VolumeUpIcon fontSize="medium" />,
    dropdown: [
      {
        component: <UploadAudio />,
      },
    ],
  },
  {
    title: "Speech text",
    icon: <ChatBubbleIcon fontSize="medium" />,
    onClick: addSpeechText,
  },
];

export default toolBarData;
