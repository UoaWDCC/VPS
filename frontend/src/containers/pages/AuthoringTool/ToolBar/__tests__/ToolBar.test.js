import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import ScenarioContextProvider from "../../../../../context/ScenarioContextProvider";
import SceneContextProvider from "../../../../../context/SceneContextProvider";
import ToolBar from "../ToolBar";

test("ToolBar component snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <ScenarioContextProvider>
        <SceneContextProvider>
          <ToolBar />
        </SceneContextProvider>
      </ScenarioContextProvider>
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
