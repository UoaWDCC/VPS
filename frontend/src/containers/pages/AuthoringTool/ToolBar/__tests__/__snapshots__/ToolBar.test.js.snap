// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar component snapshot test 1`] = `
<ul
  className="MuiList-root toolBar MuiList-padding"
  onKeyDown={[Function]}
  role="menu"
  tabIndex={-1}
>
  <div
    className="MuiBox-root MuiBox-root-1"
    onClick={[Function]}
    tabIndex={0}
  >
    <div>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text menuItem"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Image"
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text menuItem"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Text"
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.5 4v3h5v12h3V7h5V4h-13zm19 5h-9v3h3v7h3v-7h3V9z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text menuItem"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Button"
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text menuItem"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Audio"
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</ul>
`;
